### PR TITLE
Revamp and test build scripts

### LINKED ISSUE #1613: Add tests for build tool
## Feature Description

While it's somewhat verified to work OK by using it for CI steps and whatnot, there are currently no tests for the build tool in `bin/commander.js`.

Adding full test coverage would increase confidence in it.

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance Criteria

* Full or at least very hight test coverage for `bin/commander.js`

## QA Instructions

N/A

### LINKED ISSUE #1739: Fix Google Fonts download script
The script uses `import`, but is not a module. Need to replace with `require`, or turn into a module.

See https://github.com/google/web-stories-wp/runs/667122783

```
> web-stories-wp@ workflow:fonts /home/runner/work/web-stories-wp/web-stories-wp
> ./bin/build-fonts.js

/home/runner/work/web-stories-wp/web-stories-wp/bin/build-fonts.js:21
import SYSTEM_FONTS from './systemFonts';
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at wrapSafe (internal/modules/cjs/loader.js:1047:16)
    at Module._compile (internal/modules/cjs/loader.js:1097:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1153:10)
    at Module.load (internal/modules/cjs/loader.js:977:32)
    at Function.Module._load (internal/modules/cjs/loader.js:877:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12)
    at internal/main/run_main_module.js:18:47
```

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by swissspidy**
## Summary

Revamps the build tooling and makes it testable.

Adds a ton of unit tests for the build tooling, hence the many changed lines.

## Relevant Technical Choices

Bumps the project's node version to 14 and uses ECMAScript modules for node scripts, too.

Files still using CommonJS have been renamed to `cjs` because they don't yet work otherwise.

## Testing Instructions

```bash
./bin/commander.js --help

./bin/commander.js version --help

./bin/commander.js build-plugin --help

./bin/commander.js build-fonts --help
```

Verify that https://github.com/google/web-stories-wp/pull/1779/checks still lists ZIP files for the "Build plugin" workflow.

---

Fixes #1739
Fixes #1613

### CODE REVIEW

File: .github/workflows/continuous-integration-build.yml [hunk 1]
```diff
@@ -65,16 +65,16 @@ jobs:
         run: |
           echo 'assets/images/templates' >> .distignore
           npm run build:js
-          npm run workflow:build-plugin -- --cdn
-          npm run workflow:bundle-plugin web-stories-composer.zip -- --composer --skip-delete
-          npm run workflow:bundle-plugin web-stories.zip -- --skip-delete
+          npm run workflow:version -- --nightly
```
Comment (spacedmonkey) [inline .github/workflows/continuous-integration-build.yml:68]: What does change this to nightly mean?

Comment (swissspidy) [reply .github/workflows/continuous-integration-build.yml:68]: That the current commit / revision is added to the plugin version in the main plugin file.

File: bin/commander.js [hunk 2]
```diff
@@ -17,255 +17,133 @@
...
+  .option(
+    '--nightly',
+    'Whether this is a nightly build and thus should append the current revision to the version number.'
+  )
+  .description('Bump the version of the plugin')
+  .on('--help', () => {
+    console.log('');
+    console.log('Examples:');
+    console.log('  # Prepare stable release');
+    console.log('  $ commander.js version 1.2.3');
```
Comment (spacedmonkey) [inline bin/commander.js:55]: Can we define the version number in a json file or something similar and not hardcode it like this.

**Review by spacedmonkey — CHANGES_REQUESTED** [review]
This PR needs to be rebased. There is also one nit pick regarding an hard coded version number

File: bin/commander.js (hunk 2 above)
Comment (swissspidy) [reply bin/commander.js:55]: The main plugin file is currently the source of truth for the version number. If you wanna bump the version, use this command. How is that hardcoded?

Comment (swissspidy) [reply bin/commander.js:55]: For clarification, the `commander.js version 1.2.3` example here means:

_I wanna change this plugin's version from whatever it currently is to `1.2.3`_

**Review by spacedmonkey — APPROVED** [review]
LGTM

**Pull request merged**